Tools/Scripts/hooks/pre-push

```diff
@@ -0,0 +1,323 @@
+    for push in sys.stdin.readlines():
+        _, local_ref, _, remote_ref = push.split()
+        # Deletes are always safe
+        if local_ref == NULL_HASH:
+            continue
+        if remote_ref == NULL_HASH:
+            remote_ref = None
+        to_push.append((remote_ref, local_ref))
+    if not to_push:
+        if VERBOSITY > QUIET:
+            print('No novel content being pushed')
+        return 0
+
+    remote_to_name, name_to_remote = remote_name_mappings()
+    if name == remote:
+        match = REMOTE_RE.match(remote)
+        if not match:
+            print_error("'{}' does not conform to a recognized target remote".format(remote))
+            return 255
+        name = remote_to_name.get('{}:{}'.format(match.group('host'), match.group('path')), None)
+        if not name:
+            print_error("Failed to map '{}' to an existing remote".format(remote))
+            return 1
+
+
+    remotes_with = {}
+    for frm, to in to_push:
+        for commit in commits_in(frm, to):
+            if commit not in remotes_with:
+                remotes_with[commit] = remotes_for(commit) or []
+            if name in remotes_with[commit]:
+                break
+    remotes_with = dict(filter(lambda pair: name not in pair[1], remotes_with.items()))
+
+    print("Verifying commits in '{}' mode".format(MODE))
+    if VERBOSITY > QUIET:
+        print('Attempting to push {} novel commits to {}...'.format(len(remotes_with), name))
+
+    need_prompt = False
+    for commit, remotes in remotes_with.items():
+        level, klass = security_level_for_commit(commit, remotes, name_to_remote=name_to_remote)
+
+        # Unkown security level of source
+        if klass == 1 and level is None and not target_level:
+            print_error("'{}' comes from an uncategorized remote, it's not safe to push it publically".format(commit))
+            return 1
+        # Security level of source exceeds security level of target
+        if klass == 1 and level > target_level:
+            print_error("'{}' comes from a more secure remote than '{}'".format(commit, name))
+            if MODE == PUBLISH_MODE:
+                need_prompt = True
+            else:
+                return 1
+
+        # Unknown security level from original of cherry-pick
+        if klass == 2 and level is None and not target_level:
+            print_error("'{}' cherry-picks a change from an uncategorized remote, it's not safe to push it publically".format(commit))
+            return 1
+        # Security level of original of a cherry-pick exceeds security level of target
+        if klass == 2 and level > target_level:
+            print_error("'{}' cherry-picks a change from a more secure remote than '{}'".format(commit, name))
+            if MODE == DEFAULT_MODE:
+                need_prompt = True
+            else:
+                return 1
+
+        # Commit is redacted, but we're pushing it publically
+        if klass == 3 and level > target_level:
```

Comment:
Why don't we check that `level` and `target_level` are well-defined before comparing them here?